doc/api/process.markdown

```diff
@@ -59,8 +59,9 @@ finished running the process will exit. Therefore you **must** only perform
 checks on the module's state (like for unit tests). The callback takes one
 argument, the code the process is exiting with.
 
-This event may not be fired if the process terminates due to signals like
-`SIGINT`, `SIGTERM`, `SIGKILL`, and `SIGHUP`.
+This event is only emitted when node exits explicitly by process.exit() or
+implicitly by the event loop draining. It will not, for example, be fired
```

Comment:
I suggest removing the for example, for the reasons @ronkorving says. The docs say when the exit event will happen (now that you have added that). Describing various scenarios that are NOT covered by that very clear statement seems extraneous, and also subject for misinterpretation. For example, if you have a SIGINT event handler, and in the handler call `process.exit()`, are you or are you not "terminating upon receiving SIGINT"?

I understand you are trying to warn people of pitfalls, specifically one that you fell in, but this is the type of thing that should be documented in a guide section describing signal usage patterns and pitfalls (https://github.com/nodejs/docs/tree/master/src/guides), as well as the Unix process model. Its not really appropriate in the API, which is going to continue to become more reference like and more succinct as tutorial and backgrounder style information is cut from the API docs and moved into the docs repo.